src/main/java/org/wildfly/security/credential/external/PasswordClass.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2015 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.security.credential.external;
+
+import javax.security.auth.Destroyable;
+
+/**
+ * Password Class interface for classes used as external password resolvers of type 'CLASS'.
+ *
+ * Each password class specified used in {@link org.wildfly.security.credential.external.impl.ClassCredentialProvider}
+ * has to implement this interface.
+ *
+ * @author <a href="mailto:pskopek@redhat.com">Peter Skopek</a>.
+ */
+public interface PasswordClass extends Destroyable {
```

Comment:
Not sure about this. PasswordClass is interface user need to implement in so we can retrieve the password. The PasswordGuessEvidence name is confusing in this case.